### ISSUE TITLE
 Error: failed to connect to [localhost:3001]

### ISSUE THREAD

**Issue by picsoung** [meteorhacks/meteord#20]
Hi,

Thanks for this project :) I am very interested to dockerize my project, to simplify it's distribution.

I was able to create the image but when I try to run it with this command

```
docker run -d \
-e ROOT_URL=http://localhost:3000 \
-e MONGO_URL=mongodb://localhost:3001/meteor \
-e METEOR_SETTINGS='{ALL_MY_SETTINGS}' \
-p 3000:80 \
picsoung/apisio
```

but in the logs of the container I have `Error: failed to connect to [localhost:3001]`
I am just trying to run on the default port.

What goes wrong?
Thanks for your help :)

**Comment by arunoda** [comment]
You need to set a correct Mongo URL? You need to run a mongodb server on
port 3001, according to the above settings.

**Comment by picsoung** [comment]
the default mongodb server is not on 3001 when you just run your meteor application with `meteor` ?

**Comment by arunoda** [comment]
meteord is different. It don't use meteor to run your app. It's a meteor
bundle running with pure node. It don't have mongodb as well.

**Comment by picsoung** [comment]
I see... so I have to add instructions to launch the mongodb server ?
Where should it go?
on the Dockerfile?

**Comment by markshust** [comment]
any update on this? trying to just run a standard meteor app. do we have to deploy a second container for mongo?

**Comment by ryneeverett** [comment]
@picsoung @markoshust Yes, you need to run mongo separately. For an example of how to do this with docker, see the example in the comment at the end of [the docker registry page](https://registry.hub.docker.com/u/meteorhacks/meteord/).

**Comment by markshust** [comment]
Thanks, I get it now. The 'meteor' command caused us not to think of how much node is involved and how it runs meteor. This article explained quite a bit for me - http://www.qasimzeeshan.com/how-to-deploy-meteor-telescope-on-centos/

This ticket can probably be closed because it's not an issue.

**Comment by levino** [comment]
Just to clarify: Meteor runs its own "fake" mongodb when in development mode. This is not a production mongodb. You need to deploy and connect a proper mongod instance on your service / network. You can then tell Meteor to connect via network or locally to the mongodb. If you choose the latter you need to link the mongo container to the meteor container.
